Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -2772,6 +2772,44 @@ bool CSSPropertyParser::consumeWhiteSpaceShorthand(bool important)
     return true;
 }
 
+
+bool CSSPropertyParser::consumeAnimationRangeShorthand(bool important)
+{
+    // FIXME: Add extra handling for animation range sequences.
+    RefPtr start = consumeAnimationRangeStart(m_range, m_context);
+    if (!start)
+        return false;
+
+    RefPtr<CSSValue> end;
+    m_range.consumeWhitespace();
+    if (m_range.atEnd()) {
+        // From the spec: If <'animation-range-end'> is omitted and <'animation-range-start'> includes a component, then
+        // animation-range-end is set to that same and 100%. Otherwise, any omitted longhand is set to its initial value.
+
+        auto rangeEndValueForStartValue = [](const CSSValue& value) {
+            if (is<CSSPrimitiveValue>(value) && SingleTimelineRange::isOffsetValue(downcast<CSSPrimitiveValue>(value)))
+                return CSSPrimitiveValue::create(CSSValueNormal);
+            return CSSPrimitiveValue::create(value.valueID());
+        };
+
+        if (RefPtr primitiveStart = dynamicDowncast<CSSPrimitiveValue>(start))
+            end = rangeEndValueForStartValue(*primitiveStart);
+        else {
+            RefPtr pairStart = downcast<CSSValuePair>(start);
```

Comment:
Can we name these startPrimitiveValue / startPair? 

Putting start at the end makes it seems we're getting the start of the primitive value, rather than the primitive value of the start.

Reply:
Fixed.